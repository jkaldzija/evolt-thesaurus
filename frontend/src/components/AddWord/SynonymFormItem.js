import React from 'react';
import {Button, Input} from "reactstrap";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';
import ThemeButton from "../Form/ThemeButton/ThemeButton";
import classnames from "classnames";
import style from "./style.module.css";

type SynonymFormItemProps = {
    value: string,
    onChange: (string) => void,
    onDelete: () => void
}

const SynonymFormItem = ({value, onChange, onDelete}: SynonymFormItemProps) => (
    <div className="synonym-form-item d-flex mb-2">
        <Input
               className={classnames("mr-3", style.input)}
               value={value}
               required={true}
               placeholder={"Enter a synonym"}
               onChange={e => onChange(e.target.value)}/>
        <ThemeButton className={classnames(style.discard, "bg-danger text-white border-0")} onClick={onDelete}><FontAwesomeIcon icon={faTimes}></FontAwesomeIcon></ThemeButton>
    </div>
);

export default SynonymFormItem;
