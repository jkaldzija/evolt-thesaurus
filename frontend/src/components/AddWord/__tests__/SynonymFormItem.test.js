import React from 'react';
import {mount, shallow} from "enzyme";
import SynonymFormItem from "../SynonymFormItem";

describe('Synonym Form Item', () => {
    it('should render correctly', function () {
        const component = shallow(
            <SynonymFormItem value="test" onDelete={() => {}} onChange={() => {}}/>
        );
        expect(component).toMatchSnapshot();
    });
    it('should handle text input', function () {
        const changeCallback = jest.fn();
        const component = mount(
            <SynonymFormItem value="test" onDelete={() => {}} onChange={changeCallback}/>
        );
        component.find("input").simulate('change', {target: {value: "test"}});
        expect(changeCallback).toHaveBeenCalledWith("test")
    });
    it('should handle delete', function () {
        const deleteCallback = jest.fn();
        const component = mount(
            <SynonymFormItem value="test" onDelete={deleteCallback} onChange={() => {}}/>
        );
        component.find("button").simulate('click');
        expect(deleteCallback).toHaveBeenCalled();
    });
});
