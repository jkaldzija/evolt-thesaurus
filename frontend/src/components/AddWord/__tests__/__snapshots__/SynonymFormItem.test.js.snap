// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Synonym Form Item should render correctly 1`] = `
<div
  className="synonym-form-item d-flex mb-2"
>
  <Input
    className="mr-3 input"
    onChange={[Function]}
    placeholder="Enter a synonym"
    type="text"
    value="test"
  />
  <ThemeButton
    className="discard bg-danger text-white border-0"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            352,
            512,
            Array [],
            "f00d",
            "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
          ],
          "iconName": "times",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </ThemeButton>
</div>
`;
